### PR TITLE
Data context

### PR DESCRIPTION
**Pull request by SukanyaSen530**
- Data Context - Context for storing videos, liked videos, watch later and history.
- Reducer logic for storing the data
- Constants
- API Calls

### CODE REVIEW

File: src/context/actions/dataActions.js [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+};
+
+export const likeConstants = {
+  LOADING: "LIKED_LOADING",
+  GET_LIKED: "GET_LIKED",
+  ERROR: "LIKE_ERROR",
+  ADD_TO_LIKED: "ADD_TO_LIKED",
+  REMOVE_FROM_LIKED: "REMOVE_FROM_LIKED",
+};
+
```
Comment (pritam-kr) [inline src/context/actions/dataActions.js:15]: Action variable separated for every reducer??

File: src/context/providers/DataProvider.js [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+import dataReducer from "../reducers/dataReducer";
+
+const dataContext = createContext();
+
+const initialState = {
+  vid: {
+    loading: false,
+    items: [],
+    error: null,
+    single_video: {},
```
Comment (pritam-kr) [inline src/context/providers/DataProvider.js:12]: Can you explain why you keep states for loading and error? as a initial state

File: src/services/history.js [hunk 3]
```diff
@@ -0,0 +1,85 @@
+import axios from "axios";
+
+import { historyConstants } from "../context";
+
+// History
+export const loadHistoryVideos = async (dispatch) => {
+  try {
+    dispatch({ type: historyConstants.LOADING });
+    const response = await axios.get(
+      `${process.env.REACT_APP_URL}/user/history`,
```
Comment (pritam-kr) [inline src/services/history.js:10]: Also, I did get this, what is why .env here?

**Comment by pritam-kr** [comment]
Hey, I have added one or two points If you have time please explain.

BTW,
- code looks clean, readble

**Review by pritam-kr — APPROVED** [review]

**Review by sharathnair9999 — APPROVED** [review]
The code looks neat and to the point. Good work

File: src/services/history.js (hunk 3 above)
Comment (SukanyaSen530) [reply src/services/history.js:10]: Will later replace it with my backend URL.

File: src/context/providers/DataProvider.js (hunk 2 above)
Comment (SukanyaSen530) [reply src/context/providers/DataProvider.js:12]: If I keep a single loading, all pages will show loading - history, liked, playlist, that's why I separated the loading and error states.

File: src/context/actions/dataActions.js (hunk 1 above)
Comment (SukanyaSen530) [reply src/context/actions/dataActions.js:15]: I have one reducer, different action objects according to the action to be performed. Please check

**Pull request merged**